schemas.ts

```diff
@@ -0,0 +1,172 @@
+ *
+ * @param resource
+ * @param resourceSchemaName
+ * @returns The setResult schema
+ */
+const getSetResultSchema = (resource: any, resourceSchemaName: string) => {
+  const linksSchema = resource.paginate ? {
+    links: {
+      allOf: [
+        {
+          $ref: '#/components/schemas/SelfLink',
+        },
+        {
+          $ref: '#/components/schemas/PaginationLinks',
+        },
+      ],
+    },
+    meta: {
+      $ref: '#/components/schemas/Meta',
+    },
+  } : {
+    links: {
+      $ref: '#/components/schemas/SelfLink',
+    },
+  };
+  const setResultSchema: any = {
+    properties: {
+      ...linksSchema,
+      data: {
+        type: 'array',
+        items: {
+          $ref: `#/components/schemas/${resourceSchemaName}`,
+        },
+      },
+    },
+  };
+  return setResultSchema;
+};
+
+/**
+ * Gets the requestBody schema object for a resource
+ *
+ * @param resource
+ * @param resourceSchemaName
+ * @param post - true if generating a post request body, false if generating a patch request body
+ * @returns The requestBody schema
+ */
+const getRequestBodySchema = (resource: any, resourceSchemaName: string, post: boolean): any => {
+  const refPropertiesPrefix = `#/components/schemas/${resourceSchemaName}/properties`;
+  const attributeProperties = _.mapValues(resource.attributes, (_attribute, attributeName) => ({
+    $ref: `${refPropertiesPrefix}/attributes/properties/${attributeName}`,
+  }));
+  const requiredProperties = resource.requiredPostAttributes === 'all'
+    ? _.keys(resource.attributes)
+    : resource.requiredPostAttributes;
+
+  const requestBodySchema = {
+    properties: {
+      data: {
+        type: 'object',
+        properties: {
+          type: {
+            $ref: `${refPropertiesPrefix}/type`,
+          },
+          attributes: post ? {
+            type: 'object',
+            properties: attributeProperties,
+            required: requiredProperties,
+            additionalProperties: false,
+          } : {
+            type: 'object',
+            properties: attributeProperties,
+          },
+        },
+        required: post ? [
+          'type',
+          'attributes',
+        ] : [
+          'type',
+          'id',
+        ],
+      },
+    },
+    required: [
+      'data',
+    ],
+    additionalProperties: false,
+  };
+  return requestBodySchema;
+};
+
+/**
+ * Gets the post body schema object for a resource
+ *
+ * @param resource
+ * @param resourceSchemaName
+ * @returns The post body schema
+ */
+const getPostBodySchema = (resource: any, resourceSchemaName: string) => (
+  getRequestBodySchema(resource, resourceSchemaName, true)
+);
+
+/**
+ * Gets the patch body schema object for a resource
+ *
+ * @param resource
+ * @param resourceSchemaName
+ * @returns The patch body schema
+ */
+const getPatchBodySchema = (resource: any, resourceSchemaName: string) => (
+  getRequestBodySchema(resource, resourceSchemaName, false)
+);
```

Comment:
These functions are almost exactly the same. Do we need either of them?

Reply:
I'm not sure actually. My reasoning behind doing something like this was to expose these two functions and keep `getRequestBodySchema` private to simplify the calls in other modules. Now that I think about it, I think the weird convention of having the boolean `post` parameter was a reason for this. Once I fix that parameter, I think it would be fine to expose that method only and remove these two.